python/golem_task_api/client.py

```diff
@@ -79,6 +80,26 @@ def running(self) -> bool:
         """
         pass
 
+    @abc.abstractmethod
+    async def stop(self) -> None:
+        """ Force service shutdown. E.g. by calling `docker stop` """
+
+    @asynccontextmanager
+    async def started_ctx(self, command: str, port: int) -> Channel:
+        """
+        Start the service and wait for it to start listening. Yield GRPC channel
+        for connecting to the service. In case any exception is raised stop the
+        service.
+        """
+        try:
+            host, port = await self.start(command, port)
+            await wait_until_socket_open(host, port)
+            yield Channel(host, port, loop=asyncio.get_event_loop())
```

Comment:
I don't get that - why is this a context manager? Isn't this equivalent to making this a `return` statement and not being a context manager?

Reply:
I want the `except` block to get executed on error.

Reply:
Yeah, it will if it's not a contextmanager as well.

Reply:
Nevermind, brainfreeze.